### PR TITLE
Implement consistent behavior in Windows with regard to swap parameter.

### LINKED ISSUE #377: Parameter swap is ignored put_obj in windows fs backend
# Environment

**Delta-rs version**: all

**Binding**:  rust

**Environment**:
- **OS**: windows

***
# Bug

**What happened**:

`swap` parameter is ignored in windows backend `put_obj` function with changes introduced in #376

### PR DESCRIPTION
**Pull request by zijie0**
# Description

Try to implement the same behavior of swap parameter in Windows.

# Related Issue(s)

- closes #377 

# Documentation

### CODE REVIEW

**Comment by zijie0** [comment]
How to re-trigger the tests? It seems to be a network issue here.

**Comment by houqp** [comment]
@zijie0 i restarted the test for you. Going forward, you could use `git commit --amend` locally to change the commit hash, then do a force git push to the PR branch to retrigger the CI.

**Comment by zijie0** [comment]
@houqp Thanks for the tip!

File: rust/src/storage/file/mod.rs [hunk 1]
```diff
@@ -121,7 +121,7 @@ impl StorageBackend for FileStorageBackend {
             Ok(_) => Ok(()),
             Err(e) => {
                 // If rename failed, clean up the temp file.
-                //self.delete_obj(tmp_path).await?;
+                // self.delete_obj(tmp_path).await?;
```
Comment (mosyp) [inline rust/src/storage/file/mod.rs:124]: Ugh, that's a leftover from https://github.com/delta-io/delta-rs/pull/376/files#diff-ce6ba0c9d799f33c7c8ec4b5d7f36e1c03c8451390190a348b530caf838c89dfR124, I commented it out in a test and forgot to clear

Comment (mosyp) [reply rust/src/storage/file/mod.rs:124]: @zijie0 Could you please revert that back?

Comment (zijie0) [reply rust/src/storage/file/mod.rs]: Done.

**Review by mosyp — APPROVED** [review]
Great! thank you

**Pull request merged**